operate/bridge/providers/provider.py

```diff
@@ -379,48 +381,38 @@ def execute(self, provider_request: ProviderRequest) -> None:
             from_address = provider_request.params["from"]["address"]
             wallet = self.wallet_manager.load(chain.ledger_type)
             from_ledger_api = self._from_ledger_api(provider_request)
-            tx_settler = TxSettler(
-                ledger_api=from_ledger_api,
-                crypto=wallet.crypto,
-                chain_type=Chain(provider_request.params["from"]["chain"]),
-                timeout=ON_CHAIN_INTERACT_TIMEOUT,
-                retries=ON_CHAIN_INTERACT_RETRIES,
-                sleep=ON_CHAIN_INTERACT_SLEEP,
-            )
-            tx_hashes = []
 
             for tx_label, tx in txs:
                 self.logger.info(f"[PROVIDER] Executing transaction {tx_label}.")
-                nonce = from_ledger_api.api.eth.get_transaction_count(from_address)
-                tx["nonce"] = nonce  # TODO: backport to TxSettler
-                setattr(  # noqa: B010
-                    tx_settler, "build", lambda *args, **kwargs: tx  # noqa: B023
-                )
-                tx_receipt = tx_settler.transact(
-                    method=lambda: {},
-                    contract="",
-                    kwargs={},
-                    dry_run=False,
-                )
-                self.logger.info(f"[PROVIDER] Transaction {tx_label} settled.")
-                tx_hashes.append(tx_receipt.get("transactionHash", "").to_0x_hex())
+                tx_settler = TxSettler(
+                    ledger_api=from_ledger_api,
+                    crypto=wallet.crypto,
+                    chain_type=Chain(provider_request.params["from"]["chain"]),
+                    timeout=ON_CHAIN_INTERACT_TIMEOUT,
+                    retries=ON_CHAIN_INTERACT_RETRIES,
+                    sleep=ON_CHAIN_INTERACT_SLEEP,
+                    tx_builder=lambda: {
+                        **tx,  # noqa: B023
+                        "nonce": from_ledger_api.api.eth.get_transaction_count(
+                            from_address
+                        ),
+                    },
+                ).transact()
+
+                with suppress(TimeExhausted):
+                    tx_settler.settle()
+                    self.logger.info(f"[PROVIDER] Transaction {tx_label} settled.")
 
             execution_data = ExecutionData(
                 elapsed_time=time.time() - timestamp,
                 message=None,
                 timestamp=int(timestamp),
-                from_tx_hash=tx_hashes[-1],
+                from_tx_hash=tx_settler.tx_hash,
```

Comment:
This is a somewhat bad practice. tx_settler is created in the for loop, in this case it does not cause issue because we have this `if` https://github.com/valory-xyz/olas-operate-middleware/pull/206/files#diff-bc099094f1a894b4cb73318e1a115b114de583be04f48eedcf763d33d8caa52cR361, but in the general case might be problematic if the for loop is not entered.

Reply:
Bettr to store a from_tx_hash = None before the loop and update it in the loop.

Reply:
I see, you're right. Let's address that in the next PR